src/Symfony/Component/DependencyInjection/Compiler/ResolveServiceSubscribersPass.php

```diff
@@ -35,7 +35,12 @@ protected function processValue($value, $isRoot = false)
         }
 
         $serviceLocator = $this->serviceLocator;
-        $this->serviceLocator = $value->hasTag('container.service_subscriber.locator') ? $value->getTag('container.service_subscriber.locator')[0]['id'] : null;
+        $this->serviceLocator = null;
+
+        if ($value->hasTag('container.service_subscriber.locator')) {
```

Comment:
it only needs to be added to the `UnusedTagsPass` whitelist to me

Reply:
i hesitated ... AFAIK the tag is only used as intermediary internal variable. As such it'll never be used by design and thus it's always unused ... it made me preferring the self-cleaning approach here.

Reply:
That's fair to me as is also :)